code/model/modelread.cpp

```diff
@@ -4203,11 +4350,16 @@ void find_submodel_instance_point(vec3d *outpnt, object *pship_obj, int submodel
 {
 	Assert(pship_obj->type == OBJ_SHIP);
 
+	find_submodel_instance_point(outpnt, Ship_info[Ships[pship_obj->instance].ship_info_index].model_num, Ships[pship_obj->instance].model_instance_num, submodel_num);
+}
+
+void find_submodel_instance_point(vec3d *outpnt, int model_num, int model_instance_num, int submodel_num)
+{
 	vm_vec_zero(outpnt);
 	matrix submodel_instance_matrix, rotation_matrix, inv_orientation;
 
-	polymodel_instance *pmi = model_get_instance(Ships[pship_obj->instance].model_instance_num);
-	polymodel *pm = model_get(Ship_info[Ships[pship_obj->instance].ship_info_index].model_num);
+	polymodel_instance *pmi = model_get_instance(model_instance_num);
+	polymodel *pm = model_get(model_num);
```

Comment:
Same as above, either `model_num` should be validated or the parameter should be removed.